lib/pages/main/wallet_contents/settings/wallet_connect.dart

```diff
@@ -0,0 +1,323 @@
+        });
+
+        sessionDisconnectSubscription = walletConnectService
+            .onSessionDisconnect.stream
+            .listen((SessionDelete? onData) {
+          print("Got session disconnect " + onData!.topic);
+          setState(() {
+            sessions.clear();
+            sessions
+                .addAll(walletConnectService.web3Wallet!.getActiveSessions());
+          });
+        });
+
+        walletConnectService.onSessionProposal.stream
+            .listen((SessionProposalEvent? args) {
+          print("Got session proposal");
+          print(args);
+          setState(() {
+            status = "Pending session proposal";
+            if (args != null) {
+              pairingId = args.id;
+              pairingMetadata = args.params.proposer.metadata;
+
+              //Manual parsing (without registering events and methods)
+              // if ((args.params.requiredNamespaces.keys.firstOrNull != null) &&
+              //     (args.params.requiredNamespaces.keys.firstOrNull ==
+              //         "qubic:main")) {
+              //   pairingMethods =
+              //       args.params.requiredNamespaces.entries.first.value.methods;
+              //   pairingEvents =
+              //       args.params.requiredNamespaces.entries.first.value.events;
+              // } else {
+              //Automatic parsing (with registering events and methods)
+              pairingMethods = args.params.generatedNamespaces != null
+                  ? args.params.generatedNamespaces!.entries.first.value.methods
+                  : [];
+              pairingEvents = args.params.generatedNamespaces != null
+                  ? args.params.generatedNamespaces!.entries.first.value.events
+                  : [];
+              pairingNamespaces = args.params.generatedNamespaces;
+              //}
+            }
+          });
+        });
+      });
+
+      // walletConnectService.initialize().then((value) {
+      //   setState(() {
+      //     isLoading = false;
+      //     status = "Ready for connection";
+
+      //     walletConnectService.web3Wallet!.onSessionProposal.subscribe((args) {
+      //       print("Session proposal");
+
+      //       });
+      //     });
+      //     walletConnectService.web3Wallet!.onSessionConnect.subscribe((args) {
+      //       setState(() {
+      //         status = "Session connected";
+      //       });
+      //       print("Sesion connect");
+      //       print(args);
+      //     });
+
+      //     walletConnectService.web3Wallet!.onSessionProposalError
+      //         .subscribe((args) {
+      //       print("Session proposal error");
+      //       print(args);
+      //     });
+      //   });
+      // });
+    }
+  }
+
+  List<String> getMethods(SessionData sessionData) {
+    List<String> methods = [];
+    if (sessionData.requiredNamespaces == null) {
+      return [];
+    }
+    if (sessionData.requiredNamespaces!["qubic:main"] == null) {
```

Comment:
please define a constant for "qubic:main"

Reply:
Replaced with Config.walletConnectChainId